jwst/extract_1d/soss_extract/atoca.py

```diff
@@ -129,6 +142,13 @@ def __init__(self, wave_map, trace_profile, throughput, kernels,
         self.update_wave_map(wave_map)
         self.update_trace_profile(trace_profile)
 
+        # Set the mask based on trace profiles and save
+        if mask_trace_profile is None:
+            # No mask (False everywhere)
+            log.warning('`mask_trace_profile` was not given. Not optimal.')
```

Comment:
a rather terse warning. Is there more to be said or a documentation reference?

Reply:
@AntoineDarveau thoughts? 🙂

Reply:
I agree 😅. It is not optimal because the whole detector will be model. It is more optimal (in memory and time) to model only the region of interest around the traces. So the message could say:
'`mask_trace_profile` was not given. All detector's pixels will be modeled.'